include/oneapi/dpl/pstl/execution_impl.h

```diff
@@ -105,25 +78,138 @@ template <typename _ExecutionPolicy>
 using __allow_parallel = typename __internal::__policy_traits<::std::decay_t<_ExecutionPolicy>>::__allow_parallel;
 
 template <typename _ExecutionPolicy, typename... _IteratorTypes>
-auto
-__is_vectorization_preferred(_ExecutionPolicy& __exec)
-    -> decltype(__internal::__lazy_and(__exec.__allow_vector(),
-                                       __internal::__is_random_access_iterator_t<_IteratorTypes...>()))
+constexpr auto
+__is_vectorization_preferred()
 {
-    return __internal::__lazy_and(__exec.__allow_vector(),
-                                  __internal::__is_random_access_iterator_t<_IteratorTypes...>());
+    return ::std::conjunction<decltype(::std::decay_t<_ExecutionPolicy>::__allow_vector()),
+                              __internal::__is_random_access_iterator_t<_IteratorTypes...>>();
 }
 
 template <typename _ExecutionPolicy, typename... _IteratorTypes>
-auto
-__is_parallelization_preferred(_ExecutionPolicy& __exec)
-    -> decltype(__internal::__lazy_and(__exec.__allow_parallel(),
-                                       __internal::__is_random_access_iterator_t<_IteratorTypes...>()))
+constexpr auto
+__is_parallelization_preferred()
+{
+    return ::std::conjunction<decltype(::std::decay_t<_ExecutionPolicy>::__allow_parallel()),
+                              __internal::__is_random_access_iterator_t<_IteratorTypes...>>();
+}
+
+//------------------------------------------------------------------------
+// backend selector with tags
+//------------------------------------------------------------------------
+
+#if _ONEDPL_PAR_BACKEND_TBB
```

Comment:
I would have the lines 100-108 in `parallel_backend.h` header. Because `parallel_backend.h` provides(introduces) a backend.
"execution_imp.h" should contain just polices implementation stuff.

Reply:
The `parallel_backend.h` header file looks like not good place for that. It doesn't contains `__internal` namespace.